src/Microsoft.Fast.Components.FluentUI/Components/Icon/FluentIcon.cs

```diff
@@ -201,6 +240,56 @@ private string ComposedName
         }
     }
 
+    private bool GetIconChanged()
+    {
+        return _previousIconUrl != _iconUrl;
+    }
+
+    private bool GetParametersChanged()
+    {
+        if (_previousId != Id ||
+            _previousColor != _color ||
+            _previousSlot != Slot ||
+            _previousClass != Class ||
+            _previousStyle != Style ||
+            (_previousAdditionalAttributes?.Count ?? 0) != (AdditionalAttributes?.Count ?? 0))
+        {
+            return true;
+        }
+
+        if (_previousAdditionalAttributes != null &&
```

Comment:
From rendering standpoint, dictionaries of additional attributes are equivalent if their entries with not null value are equal. My implementation doesn't handle some of the cases e.g. A has entry with key 'key1' and null value, B is null/empty or doesn't have entry with key 'key1'. So i re-worked it and extracted into extension method. The performance seems to be on par with proposed implementation.
Null check (`if (xKvp.Value is null)`) in AllRenderedAttributesInAndEqual could be replace with Where filter in foreach, but it adds allocations

```csharp
internal static class AdditionalAttributesExtensions
{
    /// <summary> Determines whether tow sets of attributes are equal when rendered. </summary>
    /// <param name="x">The compared set</param>
    /// <param name="y">The set to compare with</param>
    /// <remarks></remarks>
    /// <returns><c>true</c> if both sets render the same attributes; otherwise, <c>false</c>.</returns>
    public static bool RenderedAttributesEqual(
        this Dictionary<string, object>? x,
        Dictionary<string, object>? y)
    {
        if ((x?.Count ?? 0) == 0 && (y?.Count ?? 0) == 0)
        {
            return true;
        }

        if (x is { Count: > 0 } &&
            !x.AllRenderedAttributesInAndEqual(y))
        {
            return false;
        }

        if (y is { Count: > 0 } &&
            !y.AllRenderedAttributesInAndEqual(x))
        {
            return false;
        }

        return true;
    }

    private static bool AllRenderedAttributesInAndEqual(
        this Dictionary<string, object> x,
        Dictionary<string, object>? y)
    {
        foreach (var xKvp in x)
        {
            if (xKvp.Value is null)
            {
                continue;
            }

            if (y is null) return false;

            if (!y.TryGetValue(xKvp.Key, out var yValue) ||
                !xKvp.Value.Equals(yValue))
            {
                return false;
            }
        }

        return true;
    }
}
```

Reply:
Yes, my comment was more about Clean Code and less about performance.
For me, it's very good like this.